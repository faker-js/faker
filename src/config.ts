/**
 * The possible configuration options, that can be set.
 * This type exists to be extended for plugins via type augmentation.
 *
 * The `@default` tag is used to indicate the default value, that should be used if, the config is absent.
 */
export interface FakerConfig {
  /**
   * The function used to generate the `refDate` date instance, if not provided as method param.
   * The function must return a new valid `Date` instance for every call.
   *
   * @see [Reproducible Results](https://fakerjs.dev/guide/usage.html#reproducible-results)
   * @see faker.seed(): For generating reproducible values.
   *
   * @since 9.0.0
   *
   * @default () => new Date()
   */
  defaultRefDate?: () => Date;
}
