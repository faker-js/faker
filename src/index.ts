import { Faker } from './faker';
import allLocales from './locales';

export type {
  AnimalDefinitions,
  ColorDefinitions,
  CommerceDefinitions,
  CommerceProductNameDefinitions,
  CompanyDefinitions,
  DatabaseDefinitions,
  DateDefinitions,
  DateEntryDefinition,
  FinanceCurrencyEntryDefinitions,
  FinanceDefinitions,
  HackerDefinitions,
  InternetDefinitions,
  LocaleDefinition,
  /** @deprecated Use LocationDefinitions instead */
  LocationDefinitions as AddressDefinitions,
  LocationDefinitions,
  LoremDefinitions,
  MusicDefinitions,
  /** @deprecated Use PersonDefinitions instead */
  PersonDefinitions as NameDefinitions,
  PersonDefinitions,
  /** @deprecated Use PersonTitleDefinitions instead */
  PersonTitleDefinitions as NameTitleDefinitions,
  PersonTitleDefinitions,
  PhoneNumberDefinitions,
  ScienceDefinitions,
  SystemDefinitions,
  SystemMimeTypeEntryDefinitions,
  VehicleDefinitions,
  WordDefinitions,
} from './definitions';
export { FakerError } from './errors/faker-error';
export type { FakerOptions, UsableLocale, UsedLocales } from './faker';
export type { AnimalModule } from './modules/animal';
export type {
  Casing,
  ColorFormat,
  ColorModule,
  CSSFunction,
  CSSSpace,
  NumberColorFormat,
  StringColorFormat,
} from './modules/color';
export type { CommerceModule } from './modules/commerce';
export type { CompanyModule } from './modules/company';
export type { DatabaseModule } from './modules/database';
export type { DatatypeModule } from './modules/datatype';
export type { DateModule } from './modules/date';
export type { FinanceModule } from './modules/finance';
export type { GitModule } from './modules/git';
export type { HackerModule } from './modules/hacker';
export type { HelpersModule } from './modules/helpers';
export type { ImageModule } from './modules/image';
export type { InternetModule } from './modules/internet';
export type {
  /** @deprecated Use LocationModule instead */
  LocationModule as AddressModule,
  LocationModule,
} from './modules/location';
export type { LoremModule } from './modules/lorem';
export type { MusicModule } from './modules/music';
export { Sex } from './modules/person';
export type {
  /** @deprecated Use PersonModule instead */
  PersonModule as NameModule,
  PersonModule,
  SexType,
} from './modules/person';
export type { PhoneModule } from './modules/phone';
export type { RandomModule } from './modules/random';
export type { ChemicalElement, ScienceModule, Unit } from './modules/science';
export type { StringModule } from './modules/string';
export type { SystemModule } from './modules/system';
export type { VehicleModule } from './modules/vehicle';
export type { WordModule } from './modules/word';
export { Faker };

// since we are requiring the top level of faker, load all locales by default
export const faker: Faker = new Faker({
  locales: allLocales,
});
