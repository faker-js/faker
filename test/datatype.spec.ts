import { describe, expect, it } from 'vitest';
import { faker } from '../src';
import { seededTests } from './support/seededRuns';

const NON_SEEDED_BASED_RUN = 25;

describe('datatype', () => {
  seededTests(faker, 'datatype', (t) => {
    t.describe('number', (t) => {
      t.it('noArgs')
        .itRepeated('repeated', 5, 6)
        .it('with min', { min: -42 })
        .it('with max', { max: 69 })
        .it('with min and max', {
          min: -42,
          max: 69,
        })
        .it('with min, max and precision', {
          min: -42,
          max: 69,
          precision: 0.01,
        });
    });

    t.describe('float', (t) => {
      t.it('noArgs')
        .itRepeated('repeated', 6)
        .it('with min', { min: -42 })
        .it('with max', { max: 69 })
        .it('with min and max', { min: -42, max: 69 })
        .it('with min, max and precision', {
          min: -42,
          max: 69,
          precision: 0.0001,
        });
    });

    t.describe('datetime', (t) => {
      t.it('noArgs')
        .it('with given number', Date.parse('2001-04-03T23:21:10.773Z'))
        .it('with min', {
          min: Date.parse('1622-05-23T13:45:08.843Z'),
        })
        .it('with max', {
          max: Date.parse('2002-01-29T19:47:52.605Z'),
        })
        .it('with min and max', {
          min: Date.parse('1622-05-23T13:45:08.843Z'),
          max: Date.parse('1802-01-29T19:47:52.605Z'),
        });
    });

    t.describe('string', (t) => {
      t.it('noArgs').it('with length', 42);
    });

    t.itRepeated('uuid', 5);

    t.describe('boolean', (t) => {
      t.itRepeated('noArgs', 5)
        .it('with probability', 0.42)
        .it('with probability option', { probability: 0.13 });
    });

    t.describe('hexadecimal', (t) => {
      t.it('noArgs')
        .it('with length', { length: 42 })
        .it('with prefix', { prefix: '0x' })
        .it('with casing', { case: 'lower' })
        .it('with length, prefix, and casing', {
          length: 20,
          prefix: '0x',
          case: 'lower',
        });
    });

    t.it('json');

    t.describe('array', (t) => {
      t.it('noArgs').it('with length', 4);
    });

    t.describe('bigInt', (t) => {
      t.it('noArgs').it('with value', 42);
    });
  });

  describe(`random seeded tests for seed ${faker.seed()}`, () => {
    for (let i = 1; i <= NON_SEEDED_BASED_RUN; i++) {
      describe('number', () => {
        it('should return a random number given a maximum value as Number', () => {
          const max = 10;

          const actual = faker.datatype.number(max);

          expect(actual).toBeGreaterThanOrEqual(0);
          expect(actual).toBeLessThanOrEqual(max);
        });

        it('should return a random number given a maximum value as Object', () => {
          const options = { max: 10 };

          const actual = faker.datatype.number(options);

          expect(actual).toBeGreaterThanOrEqual(0);
          expect(actual).toBeLessThanOrEqual(options.max);
        });

        it('should return a random number given a maximum value of 0', () => {
          const options = { max: 0 };

          const actual = faker.datatype.number(options);

          expect(actual).toBe(0);
        });

        it('should return a random number given a negative number minimum and maximum value of 0', () => {
          const options = { min: -100, max: 0 };

          const actual = faker.datatype.number(options);

          expect(actual).toBeGreaterThanOrEqual(options.min);
          expect(actual).toBeLessThanOrEqual(options.max);
        });

        it('should return a random number between a range', () => {
          const options = { min: 22, max: 33 };
          for (let i = 0; i < 100; i++) {
            const actual = faker.datatype.number(options);
            expect(actual).toBeGreaterThanOrEqual(options.min);
            expect(actual).toBeLessThanOrEqual(options.max);
          }
        });

        it('should return inclusive negative max value', () => {
          let foundNegative4 = false;
          let foundNegative5 = false;

          for (let iter = 0; iter < 1000; iter++) {
            const actual = faker.datatype.number({ min: -5, max: -4 });

            if (actual === -4) {
              foundNegative4 = true;
            } else if (actual === -5) {
              foundNegative5 = true;
            }

            expect(actual).toBeGreaterThanOrEqual(-5);
            expect(actual).toBeLessThanOrEqual(-4);

            if (foundNegative4 && foundNegative5) {
              break;
            }
          }

          expect(foundNegative4).toBeTruthy();
          expect(foundNegative5).toBeTruthy();
        });

        it('provides numbers with a given precision', () => {
          const options = { min: 0, max: 1.5, precision: 0.5 };
          const results = Array.from(
            new Set(
              Array.from({ length: 50 }, () => faker.datatype.number(options))
            )
          ).sort();

          expect(results).toContain(0.5);
          expect(results).toContain(1.0);

          expect(results[0]).toBe(0);
          expect(results[results.length - 1]).toBe(1.5);
        });

        it('provides numbers with a with exact precision', () => {
          const options = { min: 0.5, max: 0.99, precision: 0.01 };
          for (let i = 0; i < 100; i++) {
            const actual = faker.datatype.number(options);
            expect(actual).toBe(Number(actual.toFixed(2)));
          }
        });

        it('should not mutate the input object', () => {
          const initialMin = 1;
          const initialPrecision = 1;
          const initialOtherProperty = 'hello darkness my old friend';
          const input: {
            min?: number;
            max?: number;
            precision?: number;
            otherProperty: string;
          } = Object.freeze({
            min: initialMin,
            precision: initialPrecision,
            otherProperty: initialOtherProperty,
          });

          expect(() => faker.datatype.number(input)).not.toThrow();
        });

        it('should throw when min > max', () => {
          const min = 10;
          const max = 9;

          expect(() => {
            faker.datatype.number({ min, max });
          }).toThrowError(`Max ${max} should be greater than min ${min}.`);
        });
      });

      describe('float', () => {
        it('should return a random float with a default precision value (0.01)', () => {
          const number = faker.datatype.float();
          expect(number).toBe(Number(number.toFixed(2)));
        });

        it('should return a random float given a precision value', () => {
          const number = faker.datatype.float(0.001);
          expect(number).toBe(Number(number.toFixed(3)));
        });

        it('should return a random number given a maximum value as Object', () => {
          const options = { max: 10 };
          expect(faker.datatype.float(options)).toBeGreaterThanOrEqual(0);
          expect(faker.datatype.float(options)).toBeLessThanOrEqual(
            options.max
          );
        });

        it('should return a random number given a maximum value of 0', () => {
          const options = { max: 0 };
          expect(faker.datatype.float(options)).toBe(0);
        });

        it('should return a random number given a negative number minimum and maximum value of 0', () => {
          const options = { min: -100, max: 0 };
          expect(faker.datatype.float(options)).toBeGreaterThanOrEqual(
            options.min
          );
          expect(faker.datatype.float(options)).toBeLessThanOrEqual(
            options.max
          );
        });

        it('should return a random number between a range', () => {
          const options = { min: 22, max: 33 };
          for (let i = 0; i < 5; i++) {
            const randomNumber = faker.datatype.float(options);
            expect(randomNumber).toBeGreaterThanOrEqual(options.min);
            expect(randomNumber).toBeLessThanOrEqual(options.max);
          }
        });

        it('provides numbers with a given precision', () => {
          const options = { min: 0, max: 1.5, precision: 0.5 };
          const results = Array.from(
            new Set(
              Array.from({ length: 50 }, () => faker.datatype.float(options))
            )
          ).sort();

          expect(results).toContain(0.5);
          expect(results).toContain(1.0);

          expect(results[0]).toBe(0);
          expect(results[results.length - 1]).toBe(1.5);
        });

        it('provides numbers with a with exact precision', () => {
          const options = { min: 0.5, max: 0.99, precision: 0.01 };
          for (let i = 0; i < 100; i++) {
            const number = faker.datatype.float(options);
            expect(number).toBe(Number(number.toFixed(2)));
          }
        });

        it('should not modify the input object', () => {
          const min = 1;
          const max = 2;
          const opts = { min, max };

          faker.datatype.float(opts);

          expect(opts.min).toBe(min);
          expect(opts.max).toBe(max);
        });
      });

      describe('datetime', () => {
        it('check validity of date and if returned value is created by Date()', () => {
          const date = faker.datatype.datetime();
          expect(date).toBeTypeOf('object');
          expect(date.getTime()).not.toBeNaN();
          expect(Object.prototype.toString.call(date)).toBe('[object Date]');
        });
      });

      describe('string', () => {
        it('should generate a string value', () => {
          const generatedString = faker.datatype.string();
          expect(generatedString).toBeTypeOf('string');
          expect(generatedString).toHaveLength(10);
        });

        it('should return empty string if negative length is passed', () => {
          const negativeValue = faker.datatype.number({ min: -1000, max: -1 });
          const generatedString = faker.datatype.string(negativeValue);
          expect(generatedString).toBe('');
          expect(generatedString).toHaveLength(0);
        });

        it('should return string with length of 2^20 if bigger length value is passed', () => {
          const overMaxValue = Math.pow(2, 28);
          const generatedString = faker.datatype.string(overMaxValue);
          expect(generatedString).toHaveLength(Math.pow(2, 20));
        });
      });

      describe('boolean', () => {
        it('generates a boolean value', () => {
          const bool = faker.datatype.boolean();
          expect(bool).toBeTypeOf('boolean');
        });

        it('generates false for probability = 0', () => {
          const bool = faker.datatype.boolean(0);
          expect(bool).toBe(false);
        });

        it('generates true for probability = 1', () => {
          const bool = faker.datatype.boolean(1);
          expect(bool).toBe(true);
        });

        it.each([-5, 0.42, 5])(
          'generates a boolean value with given probability',
          (probability) => {
            const bool = faker.datatype.boolean(probability);
            expect(bool).toBeTypeOf('boolean');
          }
        );

        it('generates a boolean value for empty options', () => {
          const bool = faker.datatype.boolean({});
          expect(bool).toBeTypeOf('boolean');
        });

        it('generates false for { probability: 0 }', () => {
          const bool = faker.datatype.boolean({ probability: 0 });
          expect(bool).toBe(false);
        });

        it('generates true for { probability: 1 }', () => {
          const bool = faker.datatype.boolean({ probability: 1 });
          expect(bool).toBe(true);
        });

        it.each([-5, 0.42, 5])(
          'generates a boolean value with given probability option',
          (probability) => {
            const bool = faker.datatype.boolean({ probability });
            expect(bool).toBeTypeOf('boolean');
          }
        );

        it('should not mutate the input object', () => {
          const filledOptions: { probability?: number } = Object.freeze({
            probability: 1,
          });
          expect(() => faker.datatype.boolean(filledOptions)).not.toThrow();

          const emptyOptions: { probability?: number } = Object.freeze({});
          expect(() => faker.datatype.boolean(emptyOptions)).not.toThrow();
        });
      });

      describe('UUID', () => {
        it('generates a valid UUID', () => {
          const UUID = faker.datatype.uuid();
          const RFC4122 =
            /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/;
          expect(UUID).toMatch(RFC4122);
        });
      });

      describe('hexadecimal', () => {
        it('generates single hex character when no additional argument was provided', () => {
          const hex = faker.datatype.hexadecimal();
          expect(hex).toMatch(/^(0x)[0-9a-f]{1}$/i);
          expect(hex.substring(2)).toHaveLength(1);
        });

        it('generates a hex string with a provided prefix', () => {
          const hex = faker.datatype.hexadecimal({ prefix: '0x' });
          expect(hex).toMatch(/^(0x)[0-9A-F]+$/i);
          expect(hex).toHaveLength(3);
        });

        it('generates a random hex string with a provided length', () => {
          const hex = faker.datatype.hexadecimal({ length: 5 });
          expect(hex).toMatch(/^(0x)[0-9a-f]+$/i);
          expect(hex.substring(2)).toHaveLength(5);
        });

        it('generates a hex string with a provided casing', () => {
          const hex = faker.datatype.hexadecimal({ case: 'lower' });
          expect(hex).toMatch(/^(0x)[0-9a-f]+$/i);
          expect(hex.substring(2)).toHaveLength(1);
        });

        it('generates a hex string with a provided prefix, length, and casing', () => {
          const hex = faker.datatype.hexadecimal({
            prefix: '0x',
            length: 7,
            case: 'upper',
          });
          expect(hex).toMatch(/^(0x)[0-9A-F]+$/i);
          expect(hex.substring(2)).toHaveLength(7);
        });
      });

      describe('json', () => {
        it('generates a valid json object', () => {
          const jsonObject = faker.datatype.json();
          expect(jsonObject).toBeTypeOf('string');
          expect(JSON.parse(jsonObject)).toBeTruthy();
        });
      });

      describe('array', () => {
        it('generates an array with passed size', () => {
          const randomSize = faker.datatype.number();
          const generatedArray = faker.datatype.array(randomSize);
          expect(generatedArray).toHaveLength(randomSize);
        });

        it('generates an array with 1 element', () => {
          const generatedArray = faker.datatype.array(1);
          expect(generatedArray).toHaveLength(1);
        });
      });

      describe('bigInt', () => {
        it('should generate a bigInt value', () => {
          const generateBigInt = faker.datatype.bigInt();
          expect(generateBigInt).toBeTypeOf('bigint');
        });

        it('should generate a big bigInt value with low delta', () => {
          const min = 999999999n;
          const max = 1000000000n;
          const generateBigInt = faker.datatype.bigInt({ min, max });
          expect(generateBigInt).toBeTypeOf('bigint');
          expect(generateBigInt).toBeGreaterThanOrEqual(min);
          expect(generateBigInt).toBeLessThanOrEqual(max);
        });

        it('should return a random bigint given a maximum value as BigInt', () => {
          const max = 10n;
          expect(faker.datatype.bigInt(max)).toBeGreaterThanOrEqual(0n);
          expect(faker.datatype.bigInt(max)).toBeLessThanOrEqual(max);
        });

        it('should return a random bigint given a maximum value as Object', () => {
          const options = { max: 10n };
          expect(faker.datatype.bigInt(options)).toBeGreaterThanOrEqual(0n);
          expect(faker.datatype.bigInt(options)).toBeLessThanOrEqual(
            options.max
          );
        });

        it('should return a random bigint given a maximum value of 0', () => {
          const options = { max: 0n };
          expect(faker.datatype.bigInt(options)).toBe(0n);
        });

        it('should return a random bigint given a negative bigint minimum and maximum value of 0', () => {
          const options = { min: -100n, max: 0n };
          expect(faker.datatype.bigInt(options)).toBeGreaterThanOrEqual(
            options.min
          );
          expect(faker.datatype.bigInt(options)).toBeLessThanOrEqual(
            options.max
          );
        });

        it('should return a random bigint between a range', () => {
          const options = { min: 22, max: 33 };
          for (let i = 0; i < 100; i++) {
            const randomBigInt = faker.datatype.bigInt(options);
            expect(randomBigInt).toBeGreaterThanOrEqual(options.min);
            expect(randomBigInt).toBeLessThanOrEqual(options.max);
          }
        });

        it('should succeed with success-rate', () => {
          const min = 0n;
          const max = 1000000000000n;
          const randomBigInt = faker.datatype.bigInt({ min, max });
          expect(randomBigInt).toBeGreaterThanOrEqual(min);
          expect(randomBigInt).toBeLessThanOrEqual(max);
        });

        it('should not mutate the input object', () => {
          const initialMin = 1n;
          const initialOtherProperty = 'hello darkness my old friend';
          const input: {
            min?: bigint;
            max?: bigint;
            otherProperty: string;
          } = Object.freeze({
            min: initialMin,
            otherProperty: initialOtherProperty,
          });

          expect(() => faker.datatype.bigInt(input)).not.toThrow();
        });
      });
    }
  });
});
