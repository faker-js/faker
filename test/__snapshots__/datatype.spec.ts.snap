// Vitest Snapshot v1

exports[`datatype > 42 > array > noArgs 1`] = `
[
  "ky2eiXX/J/",
  "&Kq@X.b]\\"&",
  72199,
  93855,
  "!1}2Z=YQ!I",
  "<QYF-%<{C6",
  ")jZ3DP|XL%",
  60754,
  17052,
  "'\\"yxzUlD=\\"",
]
`;

exports[`datatype > 42 > array > with length 1`] = `
[
  "ky2eiXX/J/",
  "&Kq@X.b]\\"&",
  72199,
  93855,
]
`;

exports[`datatype > 42 > bigInt > noArgs 1`] = `379177551410048n`;

exports[`datatype > 42 > bigInt > with value 1`] = `37n`;

exports[`datatype > 42 > boolean > noArgs 1`] = `true`;

exports[`datatype > 42 > boolean > noArgs 2`] = `false`;

exports[`datatype > 42 > boolean > noArgs 3`] = `false`;

exports[`datatype > 42 > boolean > noArgs 4`] = `true`;

exports[`datatype > 42 > boolean > noArgs 5`] = `false`;

exports[`datatype > 42 > boolean > with probability 1`] = `true`;

exports[`datatype > 42 > boolean > with probability option 1`] = `false`;

exports[`datatype > 42 > datetime > noArgs 1`] = `2031-03-14T21:33:22.114Z`;

exports[`datatype > 42 > datetime > with given number 1`] = `1994-03-20T17:23:00.629Z`;

exports[`datatype > 42 > datetime > with max 1`] = `1994-07-11T09:43:47.230Z`;

exports[`datatype > 42 > datetime > with min 1`] = `1801-04-11T15:13:06.330Z`;

exports[`datatype > 42 > datetime > with min and max 1`] = `1689-09-09T08:39:09.444Z`;

exports[`datatype > 42 > float > noArgs 1`] = `37453.64`;

exports[`datatype > 42 > float > repeated 1`] = `37453.64`;

exports[`datatype > 42 > float > repeated 2`] = `79653.5`;

exports[`datatype > 42 > float > repeated 3`] = `95070.48`;

exports[`datatype > 42 > float > repeated 4`] = `18343.29`;

exports[`datatype > 42 > float > repeated 5`] = `73198.66`;

exports[`datatype > 42 > float > repeated 6`] = `77968.32`;

exports[`datatype > 42 > float > with max 1`] = `25.84`;

exports[`datatype > 42 > float > with min 1`] = `37411.64`;

exports[`datatype > 42 > float > with min and max 1`] = `-0.43`;

exports[`datatype > 42 > float > with min, max and precision 1`] = `-0.4261`;

exports[`datatype > 42 > hexadecimal > noArgs 1`] = `"0x8"`;

exports[`datatype > 42 > hexadecimal > with casing 1`] = `"0x8"`;

exports[`datatype > 42 > hexadecimal > with length 1`] = `"0x8BE4ABdd39321aD7d3fe01FfCE404F4d6db0906bd8"`;

exports[`datatype > 42 > hexadecimal > with length, prefix, and casing 1`] = `"0x8be4abdd39321ad7d3fe"`;

exports[`datatype > 42 > hexadecimal > with prefix 1`] = `"0x8"`;

exports[`datatype > 42 > json 1`] = `"{\\"foo\\":\\"ky2eiXX/J/\\",\\"bar\\":\\"&Kq@X.b]\\\\\\"&\\",\\"bike\\":72199,\\"a\\":93855,\\"b\\":\\"!1}2Z=YQ!I\\",\\"name\\":\\"<QYF-%<{C6\\",\\"prop\\":\\")jZ3DP|XL%\\"}"`;

exports[`datatype > 42 > number > noArgs 1`] = `37454`;

exports[`datatype > 42 > number > repeated 1`] = `2`;

exports[`datatype > 42 > number > repeated 2`] = `5`;

exports[`datatype > 42 > number > repeated 3`] = `6`;

exports[`datatype > 42 > number > repeated 4`] = `1`;

exports[`datatype > 42 > number > repeated 5`] = `5`;

exports[`datatype > 42 > number > with max 1`] = `26`;

exports[`datatype > 42 > number > with min 1`] = `37412`;

exports[`datatype > 42 > number > with min and max 1`] = `-1`;

exports[`datatype > 42 > number > with min, max and precision 1`] = `-0.43`;

exports[`datatype > 42 > string > noArgs 1`] = `"Cky2eiXX/J"`;

exports[`datatype > 42 > string > with length 1`] = `"Cky2eiXX/J/*&Kq@X.b]\\"&{dnx4!1}2Z=YQ!I#<QYF"`;

exports[`datatype > 42 > uuid 1`] = `"5cf2bc99-2721-407d-992b-a00fbdf302f2"`;

exports[`datatype > 42 > uuid 2`] = `"94980604-8962-404f-9371-c9368f970d9a"`;

exports[`datatype > 42 > uuid 3`] = `"2710fff9-c640-413a-b7a1-97d02e642ac4"`;

exports[`datatype > 42 > uuid 4`] = `"6838920f-dc7f-46ee-9be5-19380f5d6b48"`;

exports[`datatype > 42 > uuid 5`] = `"d95f4984-24c2-410f-ac63-400d3bbbcc91"`;

exports[`datatype > 1211 > array > noArgs 1`] = `
[
  45901,
  77826,
  "-}$_/\`4hHA",
  "afl\\"h^]dnw",
  "<q|p|5KWu3",
  "CZ|Jh!E=x\\"",
  42131,
  15894,
  "V<1bEQuA|p",
  "DW9F=V1(U7",
]
`;

exports[`datatype > 1211 > array > with length 1`] = `
[
  45901,
  77826,
  "-}$_/\`4hHA",
  "afl\\"h^]dnw",
]
`;

exports[`datatype > 1211 > bigInt > noArgs 1`] = `948721906162743n`;

exports[`datatype > 1211 > bigInt > with value 1`] = `8n`;

exports[`datatype > 1211 > boolean > noArgs 1`] = `false`;

exports[`datatype > 1211 > boolean > noArgs 2`] = `true`;

exports[`datatype > 1211 > boolean > noArgs 3`] = `false`;

exports[`datatype > 1211 > boolean > noArgs 4`] = `false`;

exports[`datatype > 1211 > boolean > noArgs 5`] = `true`;

exports[`datatype > 1211 > boolean > with probability 1`] = `false`;

exports[`datatype > 1211 > boolean > with probability option 1`] = `false`;

exports[`datatype > 1211 > datetime > noArgs 1`] = `2092-02-20T03:42:04.341Z`;

exports[`datatype > 1211 > datetime > with given number 1`] = `2000-06-14T02:54:42.082Z`;

exports[`datatype > 1211 > datetime > with max 1`] = `2001-03-20T11:14:25.251Z`;

exports[`datatype > 1211 > datetime > with min 1`] = `2065-11-10T19:27:20.915Z`;

exports[`datatype > 1211 > datetime > with min and max 1`] = `1789-03-26T15:44:45.219Z`;

exports[`datatype > 1211 > float > noArgs 1`] = `92851.09`;

exports[`datatype > 1211 > float > repeated 1`] = `92851.09`;

exports[`datatype > 1211 > float > repeated 2`] = `45901.06`;

exports[`datatype > 1211 > float > repeated 3`] = `89346.28`;

exports[`datatype > 1211 > float > repeated 4`] = `77826.18`;

exports[`datatype > 1211 > float > repeated 5`] = `22556.85`;

exports[`datatype > 1211 > float > repeated 6`] = `12988.8`;

exports[`datatype > 1211 > float > with max 1`] = `64.07`;

exports[`datatype > 1211 > float > with min 1`] = `92809.09`;

exports[`datatype > 1211 > float > with min and max 1`] = `61.07`;

exports[`datatype > 1211 > float > with min, max and precision 1`] = `61.0658`;

exports[`datatype > 1211 > hexadecimal > noArgs 1`] = `"0xE"`;

exports[`datatype > 1211 > hexadecimal > with casing 1`] = `"0xe"`;

exports[`datatype > 1211 > hexadecimal > with length 1`] = `"0xEaDB42F0e3f4A973fAB0AeefCE96DFCF49cD438dF9"`;

exports[`datatype > 1211 > hexadecimal > with length, prefix, and casing 1`] = `"0xeadb42f0e3f4a973fab0"`;

exports[`datatype > 1211 > hexadecimal > with prefix 1`] = `"0xE"`;

exports[`datatype > 1211 > json 1`] = `"{\\"foo\\":45901,\\"bar\\":77826,\\"bike\\":\\"-}$_/\`4hHA\\",\\"a\\":\\"afl\\\\\\"h^]dnw\\",\\"b\\":\\"<q|p|5KWu3\\",\\"name\\":\\"CZ|Jh!E=x\\\\\\"\\",\\"prop\\":42131}"`;

exports[`datatype > 1211 > number > noArgs 1`] = `92852`;

exports[`datatype > 1211 > number > repeated 1`] = `6`;

exports[`datatype > 1211 > number > repeated 2`] = `3`;

exports[`datatype > 1211 > number > repeated 3`] = `6`;

exports[`datatype > 1211 > number > repeated 4`] = `5`;

exports[`datatype > 1211 > number > repeated 5`] = `1`;

exports[`datatype > 1211 > number > with max 1`] = `64`;

exports[`datatype > 1211 > number > with min 1`] = `92810`;

exports[`datatype > 1211 > number > with min and max 1`] = `61`;

exports[`datatype > 1211 > number > with min, max and precision 1`] = `61.07`;

exports[`datatype > 1211 > string > noArgs 1`] = `"wKti5-}$_/"`;

exports[`datatype > 1211 > string > with length 1`] = `"wKti5-}$_/\`4hHA0afl\\"h^]dnwI<q|p|5KWu3/CZ|J"`;

exports[`datatype > 1211 > uuid 1`] = `"e7ec32f0-a2a3-4c65-abbd-0caabde64dfd"`;

exports[`datatype > 1211 > uuid 2`] = `"f379e325-9f7c-4064-a086-f23942b68e5f"`;

exports[`datatype > 1211 > uuid 3`] = `"d4694649-2183-4b32-8bd7-a336639d6997"`;

exports[`datatype > 1211 > uuid 4`] = `"10ab829b-742c-4a8b-b732-98d718d77069"`;

exports[`datatype > 1211 > uuid 5`] = `"7b91ce88-effb-4d1d-93bb-ad759e00b86c"`;

exports[`datatype > 1337 > array > noArgs 1`] = `
[
  "U/4:SK$>6Q",
  26194,
  "{:e=+kD)[B",
  "e|/Jqjjj!B",
  "GDWQgC2M;q",
  40502,
  44050,
  ".Gm3tRwnZ2",
  95735,
  42541,
]
`;

exports[`datatype > 1337 > array > with length 1`] = `
[
  "U/4:SK$>6Q",
  26194,
  "{:e=+kD)[B",
  "e|/Jqjjj!B",
]
`;

exports[`datatype > 1337 > bigInt > noArgs 1`] = `251225403255239n`;

exports[`datatype > 1337 > bigInt > with value 1`] = `25n`;

exports[`datatype > 1337 > boolean > noArgs 1`] = `true`;

exports[`datatype > 1337 > boolean > noArgs 2`] = `false`;

exports[`datatype > 1337 > boolean > noArgs 3`] = `true`;

exports[`datatype > 1337 > boolean > noArgs 4`] = `true`;

exports[`datatype > 1337 > boolean > noArgs 5`] = `true`;

exports[`datatype > 1337 > boolean > with probability 1`] = `true`;

exports[`datatype > 1337 > boolean > with probability option 1`] = `false`;

exports[`datatype > 1337 > datetime > noArgs 1`] = `2018-10-28T08:46:11.896Z`;

exports[`datatype > 1337 > datetime > with given number 1`] = `1992-12-13T04:13:59.232Z`;

exports[`datatype > 1337 > datetime > with max 1`] = `1993-03-02T00:10:04.335Z`;

exports[`datatype > 1337 > datetime > with min 1`] = `1747-07-16T01:19:54.159Z`;

exports[`datatype > 1337 > datetime > with min and max 1`] = `1669-06-22T01:21:21.236Z`;

exports[`datatype > 1337 > float > noArgs 1`] = `26202.2`;

exports[`datatype > 1337 > float > repeated 1`] = `26202.2`;

exports[`datatype > 1337 > float > repeated 2`] = `56052.42`;

exports[`datatype > 1337 > float > repeated 3`] = `15868.24`;

exports[`datatype > 1337 > float > repeated 4`] = `21258.55`;

exports[`datatype > 1337 > float > repeated 5`] = `27812.37`;

exports[`datatype > 1337 > float > repeated 6`] = `54307.53`;

exports[`datatype > 1337 > float > with max 1`] = `18.08`;

exports[`datatype > 1337 > float > with min 1`] = `26160.2`;

exports[`datatype > 1337 > float > with min and max 1`] = `-12.92`;

exports[`datatype > 1337 > float > with min, max and precision 1`] = `-12.9153`;

exports[`datatype > 1337 > hexadecimal > noArgs 1`] = `"0x5"`;

exports[`datatype > 1337 > hexadecimal > with casing 1`] = `"0x5"`;

exports[`datatype > 1337 > hexadecimal > with length 1`] = `"0x5c346ba075bd57F5A62B82d72AF39CBBB07a98cbA8"`;

exports[`datatype > 1337 > hexadecimal > with length, prefix, and casing 1`] = `"0x5c346ba075bd57f5a62b"`;

exports[`datatype > 1337 > hexadecimal > with prefix 1`] = `"0x5"`;

exports[`datatype > 1337 > json 1`] = `"{\\"foo\\":\\"U/4:SK$>6Q\\",\\"bar\\":26194,\\"bike\\":\\"{:e=+kD)[B\\",\\"a\\":\\"e|/Jqjjj!B\\",\\"b\\":\\"GDWQgC2M;q\\",\\"name\\":40502,\\"prop\\":44050}"`;

exports[`datatype > 1337 > number > noArgs 1`] = `26202`;

exports[`datatype > 1337 > number > repeated 1`] = `1`;

exports[`datatype > 1337 > number > repeated 2`] = `3`;

exports[`datatype > 1337 > number > repeated 3`] = `1`;

exports[`datatype > 1337 > number > repeated 4`] = `1`;

exports[`datatype > 1337 > number > repeated 5`] = `1`;

exports[`datatype > 1337 > number > with max 1`] = `18`;

exports[`datatype > 1337 > number > with min 1`] = `26160`;

exports[`datatype > 1337 > number > with min and max 1`] = `-13`;

exports[`datatype > 1337 > number > with min, max and precision 1`] = `-12.92`;

exports[`datatype > 1337 > string > noArgs 1`] = `"9U/4:SK$>6"`;

exports[`datatype > 1337 > string > with length 1`] = `"9U/4:SK$>6QX9@{:e=+kD)[B,e|/Jqjjj!BLGDWQgC"`;

exports[`datatype > 1337 > uuid 1`] = `"48234870-5389-445f-8b41-c61a52bf27dc"`;

exports[`datatype > 1337 > uuid 2`] = `"cc057669-8c53-474d-a677-226d3e8ed92f"`;

exports[`datatype > 1337 > uuid 3`] = `"fe6d8b8b-0db9-4fa2-b265-abc0a5d0ccf5"`;

exports[`datatype > 1337 > uuid 4`] = `"0b87afbd-8949-4dfb-8d04-19f4fe7458b2"`;

exports[`datatype > 1337 > uuid 5`] = `"0bcea83c-a7ea-428e-85db-d448f2b777a6"`;
